.github/workflows/build.yml

```diff
@@ -77,10 +77,23 @@ jobs:
           name: pyteal.docset
           path: docs/pyteal.docset.tar.gz
 
+  validate-release:
+    runs-on: ubuntu-20.04
+    container: python:3.10
+    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags') }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Validate Release
+        run: make fail-if-unreleasable
+
+
   upload-to-pypi:
     runs-on: ubuntu-20.04
     container: python:3.10
-    needs: ['build-test', 'run-integration-tests', 'build-docset']
+    needs: ['validate-release', 'build-test', 'run-integration-tests', 'build-docset']
```

Comment:
@tzaffi Let me know if I've understood the proposed process?

* My understanding:  The PR catches a version mismatch _after_ git tag creation, but _before_ pypi upload.
* If that's the case, I think the PR provides _some_ improvement because it more directly highlights the error.  But, I am interested in an approach (either via process or tooling) to either _prevent_ or _minimize_ errant tag creation.  I have some thoughts here more easily discussed live.
* For reference - Here's an example build failure when `setup.py` version was _not_ changed during a release:  https://github.com/algorand/pyteal/actions/runs/3387261940/jobs/5627959460.